WeatherApp/src/Components/FetchWeather.vue

```diff
@@ -0,0 +1,257 @@
+            case "Moderate or heavy freezing rain":
+            case "Freezing rain":
+            case "Patchy light rain":
+            case "Patchy light drizzle":
+            case "Light drizzle":
+            case "Light rain shower":
+            case "Light showers of ice pellets":
+            case "Light sleet":
+            case "Light sleet showers":
+              query = "Rainy Sky";
+              break;
+            case "Thunderstorm":
+            case "Moderate or heavy rain with thunder":
+            case "Patchy light rain with thunder":
+            case "Moderate or heavy snow with thunder":
+            case "Patchy light snow with thunder":
+              query = "Thunderstorm Sky";
+              break;
+            case "Snow":
+            case "Light snow":
+            case "Moderate snow":
+            case "Blizzard":
+            case "Patchy light snow":
+            case "Patchy moderate snow":
+            case "Patchy heavy snow":
+            case "Moderate or heavy snow showers":
+            case "Moderate or heavy snow in area with thunder":
+            case "Patchy light snow in area with thunder":
+            case "Moderate or heavy rain in area with thunder":
+            case "Patchy light rain in area with thunder":
+              query = "Snow Sky";
+              break;
+            default:
+              query = this.cityName;
+              break;
+          }
+
+          axios
+            .get(
+              `https://api.unsplash.com/search/photos?query=${query}&client_id=${this.unsplashApiKey}`
+            )
+            .then((response) => {
+              console.log(response);
+              this.photoUrl = response.data.results[0].urls.regular;
+              const img = new Image();
+              img.src = this.photoUrl;
+              img.addEventListener("load", () => {
+                document.querySelector(".fade-in").classList.add("loaded");
+              });
+            });
+        });
+    },
+  },
+  // watch: {
+  //   city: function (newValue, oldValue) {
+  //     this.getWeather();
+  //   },
+  // },
+  mounted() {
+    this.getWeather();
+  },
+};
+</script>
+
+<template>
+  <main>
+    <link
+      rel="stylesheet"
+      href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.15.3/css/all.min.css"
+    />
+    <div class="container">
+      <div class="panel">
+        <form id="locationInput">
+          <input
+            type="text"
+            class="search"
+            placeholder="Enter City"
+            v-model="city"
+            @keyup.enter="getWeather"
+          />
+          <button @click.prevent="getWeather" class="submit">
+            <i class="fa fa-search"></i>
+          </button>
+        </form>
+
+        <div class="cities">
+          <ul>
+            <li
+              @click="
+                city = 'New York';
+                country = 'USA';
+                getWeather();
+              "
```

Comment:
in general, e recomandat ca in template sa nu ai logica, ai putea pune bucata asta intr-o metoda